### PR TITLE
Add Ch. 12 problem

### PR DESCRIPTION
**Pull request by neha-peddinti**
Food Class
Ch. 12
Inheritance

### CODE REVIEW

**Comment by phrdang** [comment]
@neha-peddinti please ignore the flake8 saying that the list slicing is wrong. flake8 is wrong, there's a bug in rule E203 that's being fixed.

**Rules**
flake8 Rule E203: https://github.com/PyCQA/pycodestyle/issues/373
Black slicing rules: https://black.readthedocs.io/en/stable/the_black_code_style.html#slices

**flake8 GitHub**
Issue: https://github.com/PyCQA/pycodestyle/issues/373
PR: https://github.com/PyCQA/pycodestyle/pull/914

**Review by abhatia1205 — CHANGES_REQUESTED** [review]
This is a great problem! Only a few tweaks are required. First of all, when you call ```Food.__init__(self, *args)```, the ```self``` keyword is not supposed to be passed. Also, it is better coding practice to use the ```super()``` function instead of the name of the actual class, ```Food```. This being said, if the students aren't yet supposed to have learned about the ```super``` function, I totally get that too. 

Nice work!

File: 2_intermediate/chapter12/solutions/food_class.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    def eat(self):
+        print("You are eating " + self.name + ".")
+
+    def burn(self):
+        print("You burned %d calories." % self.cals)
+
+
+class JunkFood(Food):
+    def __init__(self, name, cals, protein, fat, sugar):
+        Food.__init__(self, name, cals, protein, fat)
```
Comment (abhatia1205) [inline 2_intermediate/chapter12/solutions/food_class.py]:
```suggestion
        super().__init__(name, cals, protein, fat)
```

File: 2_intermediate/chapter12/solutions/food_class.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        self.sugar = sugar
+
+
+class Meal(Food):
+    def __init__(self, name, cals, protein, fat, sodium):
+        Food.__init__(self, name, cals, protein, fat)
```
Comment (abhatia1205) [inline 2_intermediate/chapter12/solutions/food_class.py]:
```suggestion
        super().__init__(name, cals, protein, fat)
```

File: 2_intermediate/chapter12/solutions/food_class.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    JunkFood("Chips", 160, 2, 10, 0),
+]
+
+meals = [
+    Meal("Rice and beans", 400, 10, 5, 15),
+    Meal("Burrito", 350, 8, 9, 100),
+    Meal("Pizza", 500, 20, 15, 150),
+]
+
+cals = protein = fat = sugar = sodium = 0
```
Comment (abhatia1205) [inline 2_intermediate/chapter12/solutions/food_class.py]:
```suggestion
cals, protein, fat, sugar, sodium = 0,0,0,0,0
```

**Review by abhatia1205 — CHANGES_REQUESTED** [review]

**Review by srikar-eranky — APPROVED** [review]
Looks good to me!

**Pull request merged**